src/transformers/models/fast/modeling_fast.py

```diff
@@ -0,0 +1,380 @@
+    main_input_name = "pixel_values"
+
+    def _init_weights(self, module):
+        if isinstance(module, nn.Linear):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.Conv2d):
+            module.weight.data.zero_()
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.BatchNorm2d):
+            module.weight.data.fill_(1.0)
+            module.bias.data.zero_()
+
+
+class FASTNeck(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.reduce_layers = nn.ModuleList()
+        for in_channels, out_channels, kernel_size, stride in zip(
+            config.neck_in_channels,
+            config.neck_out_channels,
+            config.neck_kernel_size,
+            config.neck_stride,
+        ):
+            layer = FastRepConvLayer(in_channels, out_channels, kernel_size, stride)
+            self.reduce_layers.append(layer)
+        self.num_layers = len(self.reduce_layers)
+
+    def forward(self, hidden_states: torch.Tensor):
+        first_layer_hidden = hidden_states[0]
+        first_layer_hidden = self.reduce_layers[0](first_layer_hidden)
+        output_stages = [first_layer_hidden]
+
+        for layer_ix in range(1, self.num_layers):
+            layer_out = self.reduce_layers[layer_ix](hidden_states[layer_ix])
+            _, _, height, width = first_layer_hidden.size()
+            layer_out = F.interpolate(layer_out, size=(height, width), mode="bilinear")
+            output_stages.append(layer_out)
+
+        combined_hidden_states = torch.cat(output_stages, 1)
+        return combined_hidden_states
+
+
+class FASTHead(nn.Module):
+    def __init__(self, config: FastConfig) -> None:
+        super().__init__()
+        self.conv = FastRepConvLayer(
+            config.head_conv_in_channels,
+            config.head_conv_out_channels,
+            config.head_conv_kernel_size,
+            config.head_conv_stride,
+        )
+
+        padding = get_same_padding(config.head_final_kernel_size)
+
+        self.final_conv = nn.Conv2d(
+            config.head_final_in_channels,
+            config.head_final_out_channels,
+            kernel_size=config.head_final_kernel_size,
+            stride=config.head_final_stride,
+            padding=padding,
+            bias=config.head_final_bias,
+        )
+
+        self.pooling_size = config.head_pooling_size
+        self.pooling_1s = nn.MaxPool2d(kernel_size=self.pooling_size, stride=1, padding=(self.pooling_size - 1) // 2)
+        self.pooling_2s = nn.MaxPool2d(
+            kernel_size=self.pooling_size // 2 + 1, stride=1, padding=(self.pooling_size // 2) // 2
+        )
+
+        if config.head_dropout_ratio > 0:
+            self.dropout = nn.Dropout2d(config.head_dropout_ratio)
+        else:
+            self.dropout = None
```

Comment:
`self.dropout = nn.Dropout2d(config.head_dropout_ratio)` should also works fine if config.head_dropout_ratio=0, why do we need if/else here? But at least let make

```suggestion
        if config.head_dropout_ratio > 0:
            self.dropout = nn.Dropout2d(config.head_dropout_ratio)
        else:
            self.dropout = nn.Identity()
```